sites/all/modules/custom/practice/practice.module

```diff
@@ -0,0 +1,216 @@
+    ->fetchAllAssoc('nid'));
+
+  $nodes = node_load_multiple($nids);
+
+  if ($nodes) {
+    $form['nodes']['titles']['#tree'] = TRUE;
+    foreach ($nodes as $node) {
+      $form['nodes']['nid'][$node->nid] = array(
+        '#type' => 'html_tag',
+        '#tag' => 'h3',
+        '#value' => $node->nid,
+      );
+      $form['nodes']['titles'][$node->nid] = array(
+        '#type' => 'textfield',
+        '#default_value' => $node->title,
+      );
+      $form['nodes']['updated'][$node->nid] = array(
+        '#type' => 'html_tag',
+        '#tag' => 'h3',
+        '#value' => date('Y.m.d H:i:s', $node->changed),
+      );
+    }
+
+    $form['save_btn'] = array(
+      '#type' => 'submit',
+      '#value' => t('Save titles'),
+    );
+
+    $form['#theme'] = 'practice_form_table';
+    $form['#nids'] = serialize($nids);
+  }
+  else {
+    $form['warning'] = array(
+      '#type' => 'html_tag',
+      '#tag' => 'h2',
+      '#value' => t('Currently there are no nodes on the site'),
+    );
+  }
+
+  return $form;
+}
+
+/**
+ * Performs validation for latest nodes form: checks title length.
+ */
+function practice_latest_nodes_form_validate($form, &$form_state) {
+  $titles = &$form_state['input']['titles'];
+  foreach ($titles as $id => $title) {
+    $titles[$id] = trim($title);
+    if (!empty($titles[$id])) {
+      if (mb_strlen($titles[$id]) > PRACTICE_MAX_TITLE_LENGTH) {
+        form_set_error('titles-' . $id,
+          t('Title for node @id is longer than @length characters',
+            array(
+              '@length' => PRACTICE_MAX_TITLE_LENGTH,
+              '@id' => $id,
+            )
+          )
+        );
+      }
+    }
+    else {
+      form_set_error('titles-' . $id,
+        t('Title for node @id is empty!',
+          array(
+            '@id' => $id,
+          )
+        )
+      );
+    }
+  }
+}
+
+/**
+ * Performs latest nodes submit: updates nodes' titles.
+ */
+function practice_latest_nodes_form_submit($form, &$form_state) {
+  $titles = $form_state['input']['titles'];
+  if ($titles) {
+    $nids = array_keys($titles);
+    $nodes = node_load_multiple($nids);
+    if ($nodes) {
+      foreach ($nodes as $nid => $node) {
+        $titles[$nid] = trim($titles[$nid]);
+        if ($titles[$nid] !== $node->title) {
+          $node->title = $titles[$nid];
+          node_save($node);
+          drupal_set_message(t('Titile "@title" of node @nid updated',
+            array(
+              '@title' => $node->title,
+              '@nid' => $node->nid,
+            )
+            )
+          );
+        }
+      }
+    }
+  }
+}
+
+/**
+ * Implements hook_theme().
+ */
+function practice_theme($existing, $type, $theme, $path) {
+  return array(
+    'practice_form_table' => array(
+      'render element' => 'form',
+    ),
+  );
+}
+
+/**
+ * Implements hook_preprocess_HOOK().
+ *
+ * Prepossesses form so that it will have table markup.
+ */
+function practice_preprocess_practice_form_table(&$variables) {
```

Comment:
Move all the logic from that function to theme_practice_form_table()
The purpose of pre-process function is to alter EXISTING logic not to define new logic.